Upgrade EUI to v81.0.0

1Copenut opened the pull request:
## Summary

`@elastic/eui@80.0.0` ⏩ `@elastic/eui@81.0.0`

---

## [`81.0.0`](https://github.com/elastic/eui/tree/v81.0.0)

- Added ability to set `options.checked` to "mixed" in `EuiSelectable` ([#6774](https://github.com/elastic/eui/pull/6774))

**Bug fixes**

- Portalled components (e.g. `EuiPopover`, `EuiModal`, `EuiFlyout`) will correctly inherit text color from its nearest `EuiThemeProvider` parent. `<EuiText color="default">` is no longer needed. ([#6775](https://github.com/elastic/eui/pull/6775))

**Breaking changes**

- `EuiSelectable` no longer renders a `data-test-selected` attribute on its list items. Use the `aria-checked` property instead ([#6774](https://github.com/elastic/eui/pull/6774))
- Nested `EuiThemeProvider`s now render a wrapping `<span>` element in order to correctly set the inherited text `color` of all descendants. `<EuiText color="default">` is no longer needed. ([#6775](https://github.com/elastic/eui/pull/6775))

Commit:
Bumping EUI to version 81.0.0.

Commit:
Adding language translations for EUI v81.0.0.

Commit:
Updated tests for SR text. Updated types for EuiSelectable mixed options.

Commit:
Updated two more SR text blocks in unit tests.

Commit:
Changed EuiSelectable deprecated selector in two tests.

Commit:
Updated Jest test for new EuiSelectable ARIA attribute.

1Copenut committed:
Merge branch 'main' into eui/v-81.0.0

Commit:
Switched to getAttribute to find DOM selector in Jest integration test.

Commit:
Merge branch 'eui/v-81.0.0' of github.com:1Copenut/kibana into eui/v-81.0.0

1Copenut committed:
Merge branch 'main' into eui/v-81.0.0

1Copenut committed:
Merge branch 'main' into eui/v-81.0.0

Commit:
Updated 4 Cypress tests with EuiSelectable SR-only text.

Commit:
Merge branch 'eui/v-81.0.0' of github.com:1Copenut/kibana into eui/v-81.0.0

Commit:
Refactored SR-only string concatenation.

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

walterra commented:
I got a question regarding nested `EuiThemeProviders` (https://github.com/elastic/eui/pull/6775). It seems adding the `span` elements will potentially end up with a nesting of block elements like `div`s within those inline `span` elements. Is that something you discussed, just wondering if you considered that sort of inline/block nesting fine for this case.

stratoula committed:
Merge branch 'main' into eui/v-81.0.0

stratoula reviewed (approved):
Visualizations team changes LGTM

ashokaditya reviewed (approved):
`security_solution/public/management` changes LGTM.

cee-chen commented:
> I got a question regarding nested `EuiThemeProviders` ([elastic/eui#6775](https://github.com/elastic/eui/pull/6775)). It seems adding the `span` elements will potentially end up with a nesting of block elements like `div`s within those inline `span` elements. Is that something you discussed, just wondering if you considered that sort of inline/block nesting fine for this case.

Thanks for the super thoughtful and engaged question @walterra! The concept of "block" vs "inline" content no longer has any semantic meaning as of HTML5. It's now solely a presentational aspect of CSS ([see MDN docs](https://developer.mozilla.org/en-US/docs/Glossary/Block-level_content)), and there is no longer any validation or concern around a block element nested within an inline wrapper.

In this case, I opted for an inline `span` element because I was trying to make the wrapper as unobtrusive as possible and as flexible for as many cases as possible. For block children, the inline `span` naturally wraps around them to become 100% width. For inline children, the span also wraps around them as naturally as possible without creating a new full-width element on the page.

That being said, yes, an escape hatch exists via the new `<EuiThemeProvider wrapperProps>` prop. If you specifically want your wrapper to display as a block, or flex, or grid, or what have you, you could do something like this:

```tsx
<EuiThemeProvider wrapperProps={{ style: { display: 'block' } }}>
```

Or alternatively, if you're using the theme provider in a layout-sensitive context, e.g. as a flexbox or grid child, and the child of your `EuiThemeProvider` is a single element, you could do use `wrapperProps.cloneElement`:

```tsx
<EuiFlexGroup>
  <EuiThemeProvider colorMode="inverse" wrapperProps={{ cloneElement: true }}>
    <EuiFlexItem grow={false}> {/* only this wrapper will render */}
      // ...
    </EuiFlexItem>
  </EuiThemeProvider>
</EuiFlexGroup>
```

I'll do another pass here on this PR for usages of `<EuiThemeProvider>` to see if I can see in code where Kibana teams may need `wrapperProps.cloneElement`. Thanks for the reminder!

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui/v-81.0.0

cee-chen committed:
[EuiThemeProvider] Remove now-unnecessary `<EuiText color="default">` usages

cee-chen committed:
[EuiThemeProvider] Fix scenario where new wrapper would interfere with flex display

cee-chen committed:
[EuiThemeProvider] Remove unnecessary usage

`EuiBottomBar` already sets the color mode of all children to dark by default - no need to set it again around the button

cqliu1 reviewed (approved):
Presentation team changes LGTM 👍

walterra reviewed (approved):
ML related changes LGTM.

@cee-chen thanks for sharing all the details related to `EuiThemeProvider`, great to see this gives us all necessary flexibility for edge cases!

paul-tavares reviewed (approved):
Files owned by Defend Workloads LGTM 👍

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui/v-81.0.0
